test/e2e/framework/pod/utils.go

```diff
@@ -127,9 +127,17 @@ const DefaultNonRootUser = 1000
 // This includes setting RunAsUser for convenience, to pass the RunAsNonRoot check.
 // Tests that require a specific user ID should override this.
 func GetRestrictedPodSecurityContext() *v1.PodSecurityContext {
+	runAsUser := pointer.Int64(DefaultNonRootUser)
+	
+	# setting RunAsUser in Windows results invalid permissions being set on projected volumes
+	# https://github.com/kubernetes/kubernetes/issues/102849
+	if framework.NodeOSDistroIs("windows") {
```

Comment:
huh... I didn't expect this bit... do we not have any e2es that test non-homogenous clusters?

Reply:
I more expected to just drop RunAsUser

Reply:
We have [a test in Windows that explicitly tests that network connectivity](https://github.com/kubernetes/kubernetes/blob/ee0a070865a9f58ad6930a8f7231431f9667ef01/test/e2e/windows/hybrid_network.go#L45) work across Windows and Linux.  But otherwise, the assumption is the cluster we run tests against would either be Linux nodes or Windows nodes not both.

Reply:
Oh, ok so we should just drop the RunAsUser? (https://github.com/kubernetes/kubernetes/pull/109946#discussion_r882937710)

Reply:
If we drop it, the callers still need to set RunAsUser for a bunch of the tests to pass on linux, which is still going to run into the same issues of either breaking windows or needing to determine the node OS...

Reply:
> the callers still need to set RunAsUser for a bunch of the tests to pass on linux

is that true? I didn't think most tests were explicitly setting RunAsUser, and instead were running images that weren't built to use root by default

Reply:
Most of them use the pause image, which should be ok, but there's also a handful using busybox which doesn't set a user.